torch/functional.py

```diff
@@ -464,9 +464,10 @@ def stft(input: Tensor, n_fft: int, hop_length: Optional[int] = None,
     r"""Short-time Fourier transform (STFT).
 
     .. warning::
-        Setting :attr:`return_complex` explicitly will be required in a future
-        PyTorch release. Set it to False to preserve the current behavior or
-        True to return a complex output.
+        From version 1.8.0, :attr:`return_complex` must be given explicitly for
```

Comment:
This warning seems great. Same question about whether we should deprecate return_complex=False now, too?

Seems like we should? I don't think we intend to support float tensors as surrogates for complex tensors long term.